### PR TITLE
i18n(fr): Add `reference/api-reference.mdx` ft. @ArmandPhilippot

### PR DESCRIPTION
**Pull request by thomasbnt**
#### Description (required)
Add `reference/api-reference.mdx` with @ArmandPhilippot for the second part. 

> [!NOTE]
> ~~Do not merge now, miss the second part from Armand.~~

#### Related issues & labels (optional)

- Suggested label: i18n

### CODE REVIEW

**Comment by thomasbnt** [comment]
Pending merge at https://github.com/thomasbnt/docs/pull/1

File: src/content/docs/fr/reference/api-reference.mdx [hunk 1]
```diff
@@ -0,0 +1,1948 @@
...
+const { id } = Astro.params;
+---
+<h1>{id}</h1>
+```
+
+Voir aussi : [`params`](#params)
+
+### `Astro.request`
+
+`Astro.request` est un objet standard [Request](https://developer.mozilla.org/fr/docs/Web/API/Request). Il peut être utilisé pour obtenir le `url`, `headers`, `method`, et même le corps de la requête.
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/reference/api-reference.mdx:163]:
```suggestion
`Astro.request` est un objet standard [Request](https://developer.mozilla.org/fr/docs/Web/API/Request). Il peut être utilisé pour obtenir les propriétés `url`, `headers`, `method`, et même le corps de la requête.
```

File: src/content/docs/fr/reference/api-reference.mdx [hunk 2]
```diff
@@ -0,0 +1,1948 @@
...
+
+```astro
+<p>Reçu une requête {Astro.request.method} depuis "{Astro.request.url}".</p>
+<p>En-têtes de requête reçus : <code>{JSON.stringify(Object.fromEntries(Astro.request.headers))}</code>
+```
+
+Voir aussi : [`Astro.url`](#astrourl)
+
+:::note
+Avec l'option par défaut `output : 'static', `Astro.request.url` ne contient pas de paramètres de recherche, comme `?foo=bar`, car il n'est pas possible de les déterminer à l'avance lors des constructions statiques. Cependant, en mode `output : 'server'`, `Astro.request.url` contient les paramètres de recherche car ils peuvent être déterminés à partir d'une requête du serveur.
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/reference/api-reference.mdx:173]:
```suggestion
Avec l'option par défaut `output: 'static'`, `Astro.request.url` ne contient pas de paramètres de recherche, comme `?foo=bar`, car il n'est pas possible de les déterminer à l'avance lors des constructions statiques. Cependant, en mode `output: 'server'`, `Astro.request.url` contient les paramètres de recherche car ils peuvent être déterminés à partir d'une requête du serveur.
```
A closing `` ` `` was missing. Also, this is a code sample so no space before `:`.

File: src/content/docs/fr/reference/api-reference.mdx [hunk 3]
```diff
@@ -0,0 +1,1948 @@
...
+
+La valeur brute de la chaîne du cookie.
+
+##### `json`
+
+<p>
+**Type :** `() => Record<string, any>`
+</p>
+
+Analyse la valeur du cookie via `JSON.parse()`, retournant un objet. Lance si la valeur du cookie n'est pas un JSON valide.
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/reference/api-reference.mdx:270]:
```suggestion
Analyse la valeur du cookie via `JSON.parse()`, retournant un objet. Génère une erreur si la valeur du cookie n'est pas un JSON valide.
```

File: src/content/docs/fr/reference/api-reference.mdx [hunk 4]
```diff
@@ -0,0 +1,1948 @@
...
+
+Spécifie la valeur de l'en-tête du cookie [SameSite](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-rfc6265bis-09#section-5.4.7).
+
+##### `secure`
+
+<p>
+**Type :** `boolean`
+</p>
+
+If true, the cookie is only set on https sites.
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/reference/api-reference.mdx:362]:
```suggestion
Si c'est vrai, le cookie n'est défini que sur les sites https.
```

File: src/content/docs/fr/reference/api-reference.mdx [hunk 5]
```diff
@@ -0,0 +1,1948 @@
...
+**Type :** `(value: string) => string`
+</p>
+
+Permet de personnaliser la façon dont le cookie est sérialisé.
+
+### `Astro.redirect()`
+
+Permet de rediriger vers une autre page, et optionnellement de fournir un [code de réponse avec un statut HTTP](https://developer.mozilla.org/fr/docs/Web/HTTP/Status) comme second paramètre.
+
+Une page (et non un composant enfant) doit `return` le résultat de `Astro.redirect()` pour que la redirection ait lieu.
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/reference/api-reference.mdx:376]: In this case, I think we can replace the inline code with a French word to be more readable:
```suggestion
Une page (et non un composant enfant) doit retourner le résultat de `Astro.redirect()` pour que la redirection ait lieu.
```

File: src/content/docs/fr/reference/api-reference.mdx [hunk 6]
```diff
@@ -0,0 +1,1948 @@
...
+<Shout message="slots!">
+  {(message) => <div>{message}</div>}
+</Shout>
+
+<!-- s'affiche comme <div>SLOTS!</div> -->
+```
+
+### `Astro.self`
+
+  `Astro.self` permet aux composants Astro d'être appelés de manière récursive. Ce comportement vous permet d'afficher un composant Astro à partir de lui-même en utilisant `<Astro.self>` dans le modèle du composant. Cela peut être utile pour itérer sur de grands magasins de données et des structures de données imbriquées.
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/reference/api-reference.mdx:535]:
```suggestion
`Astro.self` permet aux composants Astro d'être appelés de manière récursive. Ce comportement vous permet d'afficher un composant Astro à partir de lui-même en utilisant `<Astro.self>` dans le modèle du composant. Cela peut être utile pour itérer sur de grands magasins de données et des structures de données imbriquées.
```

File: src/content/docs/fr/reference/api-reference.mdx [hunk 7]
```diff
@@ -0,0 +1,1948 @@
...
+- `/posts/[...page].astro` générerait les URL suivants `/posts`, `/posts/2`, `/posts/3`, etc.
+
+`paginate()` a les arguments suivants :
+- `pageSize` - Nombre d'éléments affichés par page
+- `params` - Envoi de paramètres supplémentaires pour la création d'itinéraires dynamiques
+- `props` - Envoi de props supplémentaires pour être disponible sur chaque page
+
+#### Le prop `page` de pagination
+
+La pagination va passer une propriété `page` à chaque page affichée qui représente une seule page de données dans la collection paginée. Cela inclut les données que vous avez paginées (`page.data`) ainsi que les métadonnées de la page (`page.url`, `page.start`, `page.end`, `page.total`, etc). Ces métadonnées sont utiles pour des choses comme un bouton "Page suivante" ou un message "Showing 1-10 of 100".
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/reference/api-reference.mdx:927]:
```suggestion
La pagination va passer une propriété `page` à chaque page affichée qui représente une seule page de données dans la collection paginée. Cela inclut les données que vous avez paginées (`page.data`) ainsi que les métadonnées de la page (`page.url`, `page.start`, `page.end`, `page.total`, etc). Ces métadonnées sont utiles pour des choses comme un bouton "Page suivante" ou un message "Pages 1-10 sur 100".
```

File: src/content/docs/fr/reference/api-reference.mdx [hunk 8]
```diff
@@ -0,0 +1,1948 @@
...
+```ts
+import type { CollectionEntry } from 'astro:content';
+```
+
+Le type `CollectionEntry<TCollectionName>` est un objet avec les valeurs suivantes. `TCollectionName` est le nom de la collection que vous interrogez (par exemple `CollectionEntry<'blog'>`).
+
+#### `id`
+
+**Disponible pour :** les collections utilisant `type: 'content'` ou `type: 'data'`  
+**Exemple de types:**
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/reference/api-reference.mdx:1262]:
```suggestion
**Exemple de types :**
```

File: src/content/docs/fr/reference/api-reference.mdx [hunk 9]
```diff
@@ -0,0 +1,1948 @@
...
+
+#### `render()`
+
+**Disponible pour :** les collections utilisant `type: 'content'` seulement  
+**Type :** `() => Promise<RenderedEntry>`
+
+Une fonction pour compiler un document Markdown ou MDX donné à afficher. Cela renvoie les propriétés suivantes :
+
+- `<Content />` - Un composant utilisé pour restituer le contenu du document dans un fichier Astro.
+- `headings` - Une liste générée de titres, [reflétant l'utilitaire `getHeadings()` d'Astro] (/fr/guides/markdown-content/#propriétés-exportées) sur les importations Markdown et MDX.
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/reference/api-reference.mdx:1304]:
```suggestion
- `headings` - Une liste générée de titres, [reflétant l'utilitaire `getHeadings()` d'Astro](/fr/guides/markdown-content/#propriétés-exportées) sur les importations Markdown et MDX.
```

**Review by ArmandPhilippot — CHANGES_REQUESTED** [review]
Some small changes, then I think we're good!

**Review by ArmandPhilippot — APPROVED** [review]
LGTM for me!

> [!caution]
> I translated the second part of the file so another reviewer could be useful.

**Comment by ArmandPhilippot** [comment]
@thomasbnt Some small changes are required on `api-reference.mdx` in the `Astro.redirect` section since #8923 has been merged.

**Comment by thomasbnt** [comment]
> @thomasbnt Some small changes are required on `api-reference.mdx` in the `Astro.redirect` section since #8923 has been merged.

Done!

**Comment by ArmandPhilippot** [comment]
@thomasbnt Thanks, I didn't realized before that I could suggest changes on multiple lines. 😅

File: src/content/docs/fr/reference/api-reference.mdx [hunk 10]
```diff
@@ -0,0 +1,1954 @@
...
+**Type :** `(value: string) => string`
+</p>
+
+Permet de personnaliser la façon dont le cookie est sérialisé.
+
+### `Astro.redirect()`
+
+**Type :** `(path: string, status?: number) => Response`
+
+Permet de rediriger vers une autre page, et optionnellement de fournir un [code de réponse avec un statut HTTP](https://developer.mozilla.org/fr/docs/Web/HTTP/Status) comme second paramètre.
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/reference/api-reference.mdx:376]:
```suggestion
Permet de rediriger vers une autre page, et optionnellement de fournir un [code de réponse avec un statut HTTP](https://developer.mozilla.org/fr/docs/Web/HTTP/Status#messages_de_redirection) comme second paramètre.
```

File: src/content/docs/fr/reference/api-reference.mdx [hunk 11]
```diff
@@ -0,0 +1,1954 @@
...
+
+- `/posts/[page].astro` générerait les URL suivants `/posts/1`, `/posts/2`, `/posts/3`, etc.
+- `/posts/[...page].astro` générerait les URL suivants `/posts`, `/posts/2`, `/posts/3`, etc.
+
+`paginate()` a les arguments suivants :
+- `pageSize` - Nombre d'éléments affichés par page
+- `params` - Envoi de paramètres supplémentaires pour la création d'itinéraires dynamiques
+- `props` - Envoi de props supplémentaires pour être disponible sur chaque page
+
+#### Le prop `page` de pagination
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/reference/api-reference.mdx:931]:
```suggestion
#### La prop `page` de pagination
```
or:

```suggestion
#### La propriété `page` de pagination
```

But in any case **la**, not **le**. 😉

File: src/content/docs/fr/reference/api-reference.mdx [hunk 12]
```diff
@@ -0,0 +1,1954 @@
...
+---
+import Shout from "../components/Shout.astro";
+---
+<Shout message="slots!">
+  {(message) => <div>{message}</div>}
+</Shout>
+
+<!-- s'affiche comme <div>SLOTS!</div> -->
+```
+
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/reference/api-reference.mdx:538]: The file has been updated in #8946, so we should add:
````suggestion

Les fonctions de rappel peuvent être transmises à des emplacements nommés à l'intérieur d'une balise d'élément HTML enveloppante avec un attribut `slot`. Cet élément est uniquement utilisé pour transférer la fonction de rappel à un emplacement nommé et ne sera pas rendu sur la page.

```astro
<Shout message="slots!">
  <fragment slot="message">
    {(message) => <div>{message}</div>}
  </fragment>
</Shout>
```

Utilisez un élément HTML standard pour la balise d'encapsulation ou toute balise en minuscules (par exemple `<fragment>` au lieu de `<Fragment />`) qui ne sera pas interprété comme un composant. N'utilisez pas l'élément HTML `<slot>` car il sera interprété comme un slot Astro.

````

**Review by ArmandPhilippot — APPROVED** [review]
LGTM! 👍🏽

**Pull request closed without merging**